indico/modules/events/registration/forms.py

```diff
@@ -65,3 +70,92 @@ def __init__(self, *args, **kwargs):
+        if not self.regform.moderation_enabled:
+            del self.skip_moderation
+        from_addresses = ['{} <{}>'.format(session.user.full_name, email)
+                          for email in sorted(session.user.all_emails, key=lambda x: x != session.user.email)]
+        self.email_from.choices = zip(from_addresses, from_addresses)
+        self.email_body.description = render_placeholder_info('registration-invitation-email', invitation=None)
+
+    def validate_email_body(self, field):
+        missing = get_missing_placeholders('registration-invitation-email', field.data, invitation=None)
+        if missing:
+            raise ValidationError(_('Missing placeholders: {}').format(', '.join(missing)))
+
+
+class InvitationFormNew(InvitationFormBase):
+    _invitation_fields = ('first_name', 'last_name', 'email', 'affiliation') + InvitationFormBase._invitation_fields
+    first_name = StringField(_('First name'), [DataRequired()],
+                             description=_("The first name of the user you are inviting."))
+    last_name = StringField(_('Last name'), [DataRequired()],
+                            description=_("The last name of the user you are inviting."))
+    email = EmailField(_('Email'), [DataRequired()], filters=[lambda x: x.lower() if x else x],
+                       description=_("The invitation will be sent to this address."))
+    affiliation = StringField(_('Affiliation'),
+                              description=_("The affiliation of the user you are inviting."))
+
+    @generated_data
+    def users(self):
+        return [{'first_name': self.first_name.data,
+                 'last_name': self.last_name.data,
+                 'email': self.email.data,
+                 'affiliation': self.affiliation.data}]
+
+    def validate_email(self, field):
+        if RegistrationInvitation.find(email=field.data).with_parent(self.regform).count():
+            raise ValidationError(_("There is already an invitation with this email address."))
+        if Registration.find(email=field.data).with_parent(self.regform).count():
+            raise ValidationError(_("There is already a registration with this email address."))
+
+
+class InvitationFormExisting(InvitationFormBase):
+    _invitation_fields = ('users_field',) + InvitationFormBase._invitation_fields
+    users_field = _UsersField(_('Users'), [DataRequired()], description=_("Select the users to invite."))
+
+    @generated_data
+    def users(self):
+        return [{'first_name': x['firstName'],
+                 'last_name': x['familyName'],
+                 'email': x['email'].lower(),
+                 'affiliation': x['affiliation']}
+                for x in self.users_field.data]
+
+    def validate_users_field(self, field):
+        emails = {x['email'].lower() for x in field.data}
+        # invitations
+        existing = {x.email for x in self.regform.invitations} & emails
+        if existing:
```

Comment:
Wouldn't it be better to skip existing emails instead of throwing a `ValidationError`?